manifests/charts/istiod-remote/files/injection-template.yaml

```diff
@@ -0,0 +1,414 @@
+template: |
```

Comment:
I don't think this makes sense here. Its confusing - users will think injection is happening on this cluster, when it isn't actually. `istioctl` does not actually require it - it can be configured to fetch from the remote cluster or from files. I suspect if we actually did do this and people were doing kube-inject they would often find this to be out of sync with the remote cluster and see odd behavior. Additionally, its hard to maintain.

Reply:
istioctl Kube-inject currently reads the default injection info from the cluster where user is running by default.  Are you saying we should configure the sidecar injector cm to read the injection template from the main cluster?  I think requiring not having istioctl Kube-inject working out of box and require configuring params is unintuitive.   From my testing, the istio sidecar injector cm needs to be in place in the cluster where user runs istioctl Kube-inject to get the injection info.

Reply:
It does work out of the box, at least in my opinion. 

With current istioctl:
* If I want to inject my pod using cluster1, I run `istioctl kube-inject --kubeconfig cluster1`.
* If I want to inject my pod using cluster2, I run `istioctl kube-inject --kubeconfig cluster2`.
* If I want to use files (recommended) I pass `--injectConfigFile`

In the remote cluster model, the same model remains. Say our kubeconfigs are `local` and `remote`. User would run `istioctl kube-inject --kubeconfig remote | kubectl apply --kubeconfig local`. This is accurately modeling what is happening - the remote cluster is injecting the pod, and its deployed in the local cluster, just like using auto injections.

Reply:
What if user doesn't have kubeconfig for the central istiod?  e.g. I'm dedicated a central istiod cluster for just running istio control plane.   Normal user can only access the remote control plane cluster.  In this case, I can't ask users to do

```istioctl kube-inject --kubeconfig remote | kubectl apply --kubeconfig local```

Reply:
Then what makes you think the `injection-template.yaml` in your cluster actually matches what is used in the remote cluster if you don't control the remote cluster? And why would you want to use the configmap rather than files?

Reply:
Chatted with @howardjohn offline to understand his concerns.  Basically the chart is just to lay off the initial configuration, we expect central istiod to patch/refresh the WebHooks and configmaps (Istio-sidecar-injector and possibly istio-ca-root-cert) as necessary.

Reply:
@howardjohn I am unclear if you are satisfied with this answer. If so, please click the resolve conversation button.